src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -102,21 +102,28 @@ class NativeSessionStorage implements SessionStorageInterface
      */
     public function __construct(array $options = array(), $handler = null, MetadataBag $metaBag = null)
     {
-        $options += array(
-            // disable by default because it's managed by HeaderBag (if used)
-            'cache_limiter' => '',
-            'use_cookies' => 1,
-        );
-
-        if (\PHP_VERSION_ID >= 50400) {
-            session_register_shutdown();
+        $this->setMetadataBag($metaBag);
+
+        if (PHP_VERSION_ID >= 70200 && \PHP_SESSION_ACTIVE === session_status()) {
```

Comment:
I think we should try as much as possible to NOT have a different behavior on 7.2 than before.

Reply:
This could be a BC break if we don't add this version condition as it's a behaviour changed in PHP's runtime.

Reply:
upgrading to 7.2 will hit the BC break, so this doesn't prevent it, it just hides it :)

Reply:
Fair enough. So you'd like me to remove the PHP version condition in here? Note that we'll have to add a check to know if the constant exists as well.

Reply:
should be a no-op then, which matches the behavior pre-7.2
on 3.4, we could trigger a deprecation, and throw on 4.0

Reply:
Alright, let's do this in three PRs instead of directly throwing on 2.7:
1. On 2.7, just ignore and don't set the options/handler so it won't break with the INI settings issue
2. On 3.4, trigger a deprecation
3. On master, throw an exception